src/icon-loader.c

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <assert.h>
+#include <sfdo-desktop.h>
+#include <sfdo-icon.h>
+#include <sfdo-basedir.h>
+#include <string.h>
+#include <wlr/util/log.h>
+#include "common/mem.h"
+#include "config.h"
+#include "icon-loader.h"
+#include "img/img-png.h"
+
+#if HAVE_RSVG
+#include "img/img-svg.h"
+#endif
+
+#include "labwc.h"
+
+struct icon_loader {
+	struct sfdo_basedir_ctx *basedir_ctx;
+	struct sfdo_desktop_ctx *desktop_ctx;
+	struct sfdo_icon_ctx *icon_ctx;
+	struct sfdo_desktop_db *desktop_db;
+	struct sfdo_icon_theme *icon_theme;
+};
+
+void
+icon_loader_init(struct server *server)
+{
+	struct icon_loader *loader = znew(*loader);
+	loader->basedir_ctx = sfdo_basedir_ctx_create();
```

Comment:
If you don't plan to use `basedir_ctx` for anything else (e.g. configuration loading), it can be destroyed immediately after creating desktop and icon contexts.

(I should document that...)

Reply:
Fixed.